cart/interfaces/controller/cartapicontroller.go

```diff
@@ -145,6 +145,53 @@ func (cc *CartAPIController) ApplyGiftCardAndGetAction(ctx context.Context, r *w
 	return cc.handlePromotionAction(ctx, r, "giftcard_error", cc.cartService.ApplyGiftCard)
 }
 
+// ApplyCombinedVoucherGift applies a given first to the voucher API Endpoint (if no voucher code is already set)
+// and then to to the Giftcard API Endpoint and then eventually returns the cart
+// the request needs a query string param "anyCode" which includes the corresponding code for either a promo code
+// or a gift card
+func (cc *CartAPIController) ApplyCombinedVoucherGift(ctx context.Context, r *web.Request) web.Result {
+	result := newResult()
+
+	// fetching anyCode - which might be either voucher or giftcard
+	anyCode := r.Params["anyCode"]
+
+
+	currentCart, _, err := cc.cartReceiverService.GetCart(ctx, r.Session())
+	if err != nil {
+		result.SetError(err, "combined_cart_error")
+		response := cc.responder.Data(result)
+		response.Status(500)
+
+		return response
+	}
+
+	if len(currentCart.AppliedCouponCodes) == 0 {
+		// no coupon at cart, try to apply voucher
```

Comment:
I think we would like to overwrite an existing voucher code, so we leave out this if.
Apply voucher code -> success -> return
Apply voucher code -> failure -> Apply giftcard code